### PR TITLE
Migrate to ReScript.

### PR DESCRIPTION
**Pull request by parkerziegler**
Here we go! When merged, this PR will officially migrate us from Reason to ReScript. While it does make the library name out of date :joy: I feel this change is important for the future maintenance of this project. We'll consider a name change down the road, likely in a future major.

The migration here is automated. For the curious, you can check out [my hacky Bash script](https://gist.github.com/parkerziegler/2c0a6a7524ed8122c191cd56bad6e73a) to run this 😂 . There's nothing really meaningful to review here. We'll release this as a new minor version (3.1.0) alongside #230, assuming no regressions caught in CI.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -67,15 +65,5 @@
...
-  },
-  "lint-staged": {
-    "*.{re,rei}": [
-      "bsrefmt --in-place"
-    ]
-  },
-  "husky": {
-    "hooks": {
-      "pre-commit": "lint-staged"
-    }
```
Comment (parkerziegler) [inline package.json:79]: Unfortunately, [refmt no longer has support for in-place modification 😭](https://forum.rescript-lang.org/t/reformat-rescript-in-place/207) This means we can't do any sort of pre-commit or pre-merge checks and instead just need to ask people to use the VSCode extension or run refmt themselves :oof:

Comment (kitten) [reply package.json:79]: We can likely just restore support by writing a small script 👍

**Review by kitten — APPROVED** [review]

File: examples/5-subscription/src/app/Util.res [hunk 2]
```diff
@@ -1,6 +1,6 @@
-@bs.scope("Math") @bs.val external random: unit => float = "random"
-@bs.scope("Math") @bs.val external floor: float => int = "floor"
-@bs.send external toString: (int, int) => string = "toString"
+@scope("Math") @val external random: unit => float = "random"
+@scope("Math") @val external floor: float => int = "floor"
+@send external toString: (int, int) => string = "toString"
```
Comment (kitten) [inline examples/5-subscription/src/app/Util.res:3]: No more `bs` 😆

Comment (parkerziegler) [reply examples/5-subscription/src/app/Util.res:3]: :rip: Can't say I'm sad to see it go!

**Pull request merged**